src/backend/tty.rs

```diff
@@ -455,13 +494,17 @@ impl Tty {
             })
             .unwrap();
 
+        let drm_lease_state = DrmLeaseState::new::<State>(&niri.display_handle, &node).unwrap();
```

Comment:
This shouldn't `unwrap()`. If it's expected to "always" work then I guess `.context()?` it (and move as early as possible in the function so that nothing needs to be deinitialized should this fail). If it's not expected to "always" work then make it an `Option` in the `OutputDevice`.